File: src/Insight.Box.FrontEnd/src/modules/workflows/models/FeedbackAnalysisWorkflowResultFilter.ts
import {FilterPagination} from "@/common/FilterPagination";

/*
 * Represents workflow results filter
 */
export class FeedbackAnalysisWorkflowResultFilter extends  FilterPagination {

    /*
     * Workflow id
     */
    public workflowId: string;

    constructor(workflowId: string) {
        super();
        this.pageSize = 5;
        this.workflowId = workflowId;
    }
}

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/FeedbackAnalysisWorkflowResults/Models/FeedbackAnalysisWorkflowResultFilter.cs
using Feedback.Analyzer.Application.Organizations.Models;
using Feedback.Analyzer.Domain.Common.Queries;

namespace Feedback.Analyzer.Application.FeedbackAnalysisWorkflowResults.Models;

public class FeedbackAnalysisWorkflowResultFilter : FilterPagination
{
    /// <summary>
    /// Gets or sets workflow id
    /// </summary>
    public Guid? WorkflowId { get; set; }

    public override int GetHashCode()
    {
        var hashCode = new HashCode();

        hashCode.Add(PageSize);
        hashCode.Add(PageToken);

        return hashCode.ToHashCode();
    }

    public override bool Equals(object? obj) => 
        obj is OrganizationFilter clientFilter 
        && clientFilter.GetHashCode() == GetHashCode();
}